pkg/operator/metrics/metrics.go

```diff
@@ -0,0 +1,80 @@
+package metrics
+
+import (
+	"github.com/blang/semver"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/service-ca-operator/pkg/operator/v4_00_assets"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/util/cert"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+func NewServiceCAMetrics() *ServiceCAMetrics {
+	s := &ServiceCAMetrics{
+		expiryTime: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "service_ca_expiry_time",
+			Help: "Reports the expiry time of service CA used in the cluster.",
+		}, []string{"expiry_time"}),
+		timeLeft: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "service_ca_time_left",
+			Help: "Reports the time left before the service CA expires.",
+		}, []string{"time_left"}),
+	}
+	legacyregistry.MustRegister(s)
+	return s
+}
+
+// ServiceCAMetrics implements metrics gathering for this component.
+type ServiceCAMetrics struct {
+	lock       sync.Mutex
+	expiryTime *prometheus.GaugeVec
+	timeLeft   *prometheus.GaugeVec
+}
+
+func (s *ServiceCAMetrics) Create(version *semver.Version) bool {
+	return true
+}
+
+// Describe reports the metadata for metrics to the prometheus collector.
+func (s *ServiceCAMetrics) Describe(ch chan<- *prometheus.Desc) {
+	ch <- s.expiryTime.WithLabelValues("").Desc()
+	ch <- s.timeLeft.WithLabelValues("").Desc()
+}
+
+// Collect calculates metrics from the cached config and reports them to the prometheus collector.
+func (s *ServiceCAMetrics) Collect(ch chan<- prometheus.Metric) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	ch <- s.expiryTime.WithLabelValues("")
+	ch <- s.timeLeft.WithLabelValues("")
+}
+
+func (s *ServiceCAMetrics) ClearState() {}
+
+func (s *ServiceCAMetrics) Compute(client coreclientv1.CoreV1Interface) {
```

Comment:
When I asked for the current CA duration to be collected via the sync loop, I meant in `manageSignerCA` where the CA is already being parsed: 

https://github.com/openshift/service-ca-operator/blob/master/pkg/operator/sync_common.go#L77

I don't think it should be necessary for metrics computation to independently load and parse the CA.

Reply:
moved logic to `manageSignerCA` and metrics package will not be aware of CA data.